### ISSUE TITLE
Failed to use GPU acceleration for hybrid/overlay type potential

### ISSUE THREAD

**Issue by ruio248** [lammps/lammps#4131]
**When I tried to use GPU acceleration ,it is work for lj type potential .But for hybrid/overlay type potential I encountered the following error.**

(base) usr@usr:/media/ruihao/lammps$ mpirun -np 4 lmp -sf gpu -pk gpu 1 -in in
LAMMPS (2 Aug 2023 - Update 2)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
DumpFolderName
Lattice spacing in x,y,z = 3.14 3.14 3.14
Created orthogonal box = (-188.4 -188.4 -188.4) to (188.4 188.4 188.4)
  1 by 2 by 2 MPI processor grid
Created 3456000 atoms
  using lattice units in orthogonal box = (-188.4 -188.4 -188.4) to (188.4 188.4 188.4)
  create_atoms CPU = 0.051 seconds
3162509 atoms in group NO_FIX
293491 atoms in group FIX
2698631 atoms in group FREE
463878 atoms in group BATH
1 atoms in group CENTER
WARNING: Calling write_dump before a full system init. (src/write_dump.cpp:65)

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

Your simulation uses code contributions which should be cited:
- GPU package (short-range, long-range and three-body potentials): doi:10.1016/j.cpc.2010.12.021, doi:10.1016/j.cpc.2011.10.012, doi:10.1016/j.cpc.2013.08.002, doi:10.1016/j.commatsci.2014.10.068, doi:10.1016/j.cpc.2016.10.020, doi:10.3233/APC200086
The log file lists these citations in BibTeX format.

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

ERROR: GPU package styles must not be used multiple times (src/pair_hybrid.cpp:608)
Last command: run             2000
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.

**LAMMPS Version and Platform**
Version : 2 Aug 2023 - Update 2
OS: Ubuntu 20.04 LTS
Cuda version : 11.8
Driver Version: 535.113.01
GPU： 1 RTX 4090

**Potential setting file :**
pair_style hybrid/overlay eam/alloy zbl 1.4 1.8
pair_coeff * * eam/alloy WHff.eam.alloy W H
pair_coeff 1 1 zbl 74.0 74.0 
pair_coeff 2 2 zbl 1.0 1.0
pair_coeff 1 2 zbl 74.0 1.0
mass 1 1.83840E+02
mass 2 1.00794E+00

**Comment by ndtrung81** [comment]
@ruio248 is it possible to share with us a simplified input deck with a smaller system size to reproduce the issue?

**Comment by akohlmey** [comment]
> ERROR: GPU package styles must not be used multiple times (src/pair_hybrid.cpp:608)

This error *cannot* be produced with the input you quoted since in the pair style command no pair style is listed more than once.

What you *should* get is:
```
ERROR: Must not use GPU neighbor lists with hybrid pair style (src/GPU/fix_gpu.cpp:280)
```
Which can be addressed with adding `-pk gpu 4 neigh no` to your command line.

**Comment by ruio248** [comment]
I still confront with same issue and adding -pk gpu 4 neigh no dosen't work for me. 
## input
variable        env_temp index 300 #enviroment temperature
label           loop_env_temp

variable        E_PKA index 90.2   # unit: keV
label           loop_E_PKA
variable        d_x index 0   #direction 
variable        d_y index 0 
variable        d_z index -1
variable        angle index 1
label           loop_angle

variable        random loop 1 
label           loop_random
variable        halt_dt equal -0.0000005*${env_temp}+0.00125

variable        box_r equal 60 # the number of atoms > 25 times the energy of the projectile in eV (eg: 500 000 atoms for 20 keV) 
variable        fix_th equal 2   # fix thickness
variable        bath_th equal 3   # bath thickness
variable        center equal 25  

variable        lattice_constant equal 3.14
variable        v_PKA equal 1*sqrt(${E_PKA})*324.3739594  # unit: Angstrom / ps
variable        tem_cv equal ${env_temp}*2
variable        box_l equal -1*${box_r}
variable        nf_r equal ${box_r}-${fix_th}
variable        nf_l equal -1*${nf_r}
variable        f_r equal ${nf_r}-${bath_th}
variable        f_l equal -1*${f_r}
variable        c_r equal ${center}+0.1 
variable        c_l equal ${center}-0.1 
variable        diagonal equal sqrt(${d_x}*${d_x}+${d_y}*${d_y}+${d_z}*${d_z})
variable        vx equal ${v_PKA}/${diagonal}*${d_x}
variable        vy equal ${v_PKA}/${diagonal}*${d_y}
variable        vz equal ${v_PKA}/${diagonal}*${d_z}

units           metal
atom_style      atomic
atom_modify     map array
boundary        p p p 
lattice         bcc ${lattice_constant}

region          box block ${box_l} ${box_r} ${box_l} ${box_r} ${box_l} ${box_r}
create_box      2 box 
create_atoms    1 box
region          NO_FIX block ${nf_l} ${nf_r} ${nf_l} ${nf_r} ${nf_l} ${nf_r}
group           NO_FIX region NO_FIX
group           FIX subtract all NO_FIX
region          FREE block ${f_l} ${f_r} ${f_l} ${f_r} ${f_l} ${f_r}
group           FREE region FREE
group           BATH subtract NO_FIX FREE
region          CENTER block -0.1 0.1 -0.1 0.1 ${c_l} ${c_r}
group           CENTER region CENTER

log             log/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}.log.lammps
include         pot
thermo          100
thermo_style    custom step time dt temp
variable        DT equal dt
variable        TEM equal temp
variable        TEMP_DT equal ${DT}*100
compute         1 all pe/atom
compute         2 all voronoi/atom occupation
compute         3 all ke/atom


shell           mkdir dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}
write_dump      all custom dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}/ref.dump id type x y z 


fix             FIXATOMS FIX setforce 0.0 0.0 0.0
velocity        NO_FIX create ${tem_cv} ${random} dist gaussian mom yes rot yes 
fix             1 all nvt temp ${env_temp} ${env_temp} 0.01 drag 0.2
timestep        0.0005
run             2000
unfix           1

 incident 
reset_timestep  0
velocity        CENTER set ${vx} ${vy} ${vz} sum yes units box    # adjust here if change direction of PKA
fix             1 BATH nvt temp ${env_temp} ${env_temp} ${TEMP_DT} drag 0.2
fix             2 FREE nve 
dump            1 all custom 100 dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}/lammps_iv.dump id type c_2[1]
dump_modify     1 thresh c_2[1] != 1
dump            2 all custom 100 dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}/lammps_2i.dump id type x y z  
dump_modify     2 thresh c_2[2] > 1
dump            3 all custom 100 dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}/high_pe.dump id type x y z c_1
dump_modify     3 thresh c_1 > -8.6
dump            4 all custom 1000 dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}/all.dump id type x y z c_1
fix             3 all dt/reset 10 NULL NULL 0.01 units box 
fix             4 all halt 10 v_DT > ${halt_dt} error continue         # adjust here for end criterion
run             100000000
unfix           4
unfix           3
timestep        0.001
run             40000
#undump          4
undump          3
undump          2
undump          1
unfix           2
unfix           1

min_style       cg
minimize        1e-14 1e-14 1000 1000
dump            1 all custom 1 dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}/lammps_iv_last.dump id type c_2[1]
dump_modify     1 thresh c_2[1] != 1
dump            2 all custom 1 dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}/lammps_2i_last.dump id type x y z  
dump_modify     2 thresh c_2[2] > 1
dump            3 all custom 1 dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}/high_pe_last.dump id type x y z c_1
dump_modify     3 thresh c_1 > -8.6
dump            4 all custom 1 dump/${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}/all_last.dump id type x y z c_1 c_2[1] c_2[2] 
run             0
undump          4
undump          3
undump          2
undump          1

print           "${env_temp}K_${E_PKA}KeV_[${d_x}${d_y}${d_z}]_${random}" append folder_name.csv

clear 
next            random
jump            SELF loop_random               #  random number
next            d_x
next            d_y
next            d_z
next            angle
jump            SELF loop_angle         
next            E_PKA
jump            SELF loop_E_PKA           
next            env_temp
jump            SELF loop_env_temp           

print           "Program End!"
## potential  setting file 
pair_style hybrid/overlay eam/alloy zbl 1.4 1.8
pair_coeff * * eam/alloy WHff.eam.alloy W H
pair_coeff 1 1 zbl 74.0 74.0 
pair_coeff 2 2 zbl 1.0 1.0
pair_coeff 1 2 zbl 74.0 1.0
mass 1 1.83840E+02
mass 2 1.00794E+00
## issue 
### gpu -pk 4
(base) ruihao@ruihao:/media/ruihao/lammps$ mpirun -np 4 lmp -sf gpu -pk gpu 4 neigh no -in in
LAMMPS (2 Aug 2023 - Update 2)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
ERROR: Could not find/initialize a specified accelerator device (src/GPU/gpu_extra.h:57)
Last command: package gpu 4 neigh no
### gpu -pk 1
(base) ruihao@ruihao:/media/ruihao/lammps$ mpirun -np 4 lmp -sf gpu -pk gpu 1 neigh no -in in
LAMMPS (2 Aug 2023 - Update 2)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
DumpFolderName
Lattice spacing in x,y,z = 3.14 3.14 3.14
Created orthogonal box = (-188.4 -188.4 -188.4) to (188.4 188.4 188.4)
  1 by 2 by 2 MPI processor grid
Created 3456000 atoms
  using lattice units in orthogonal box = (-188.4 -188.4 -188.4) to (188.4 188.4 188.4)
  create_atoms CPU = 0.072 seconds
3162509 atoms in group NO_FIX
293491 atoms in group FIX
2698631 atoms in group FREE
463878 atoms in group BATH
1 atoms in group CENTER
WARNING: Calling write_dump before a full system init. (src/write_dump.cpp:65)

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

Your simulation uses code contributions which should be cited:
- GPU package (short-range, long-range and three-body potentials): doi:10.1016/j.cpc.2010.12.021, doi:10.1016/j.cpc.2011.10.012, doi:10.1016/j.cpc.2013.08.002, doi:10.1016/j.commatsci.2014.10.068, doi:10.1016/j.cpc.2016.10.020, doi:10.3233/APC200086
The log file lists these citations in BibTeX format.

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

ERROR: GPU package styles must not be used multiple times (src/pair_hybrid.cpp:608)
Last command: run             2000
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.

**Comment by ruio248** [comment]
I still face the same problem. Adding -pk gpu 4 neigh no seems not works for me ,the related file have uploaded
[test_code.zip](https://github.com/lammps/lammps/files/15103106/test_code.zip)
## issue 1 
(base) ruihao@ruihao:/media/ruihao/lammps$ mpirun -np 4 lmp -sf gpu -pk gpu 1 neigh no -in in
LAMMPS (2 Aug 2023 - Update 2)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
DumpFolderName
Lattice spacing in x,y,z = 3.14 3.14 3.14
Created orthogonal box = (-188.4 -188.4 -188.4) to (188.4 188.4 188.4)
  1 by 2 by 2 MPI processor grid
Created 3456000 atoms
  using lattice units in orthogonal box = (-188.4 -188.4 -188.4) to (188.4 188.4 188.4)
  create_atoms CPU = 0.072 seconds
3162509 atoms in group NO_FIX
293491 atoms in group FIX
2698631 atoms in group FREE
463878 atoms in group BATH
1 atoms in group CENTER
WARNING: Calling write_dump before a full system init. (src/write_dump.cpp:65)

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

Your simulation uses code contributions which should be cited:
- GPU package (short-range, long-range and three-body potentials): doi:10.1016/j.cpc.2010.12.021, doi:10.1016/j.cpc.2011.10.012, doi:10.1016/j.cpc.2013.08.002, doi:10.1016/j.commatsci.2014.10.068, doi:10.1016/j.cpc.2016.10.020, doi:10.3233/APC200086
The log file lists these citations in BibTeX format.

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

ERROR: GPU package styles must not be used multiple times (src/pair_hybrid.cpp:608)
Last command: run             2000
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 429.
Cuda driver error 4 in call at file '/home/ruihao/lammps/lammps-2Aug2023/lib/gpu/geryon/nvd_device.h' in line 430.
## issue 2 
(base) ruihao@ruihao:/media/ruiha/lammps$ mpirun -np 4 lmp -sf gpu -pk gpu 4 neigh no -in in
LAMMPS (2 Aug 2023 - Update 2)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
ERROR: Could not find/initialize a specified accelerator device (src/GPU/gpu_extra.h:57)
Last command: package gpu 4 neigh no
